libtransmission/bandwidth.cc

```diff
@@ -164,19 +164,24 @@ void tr_bandwidth::allocateBandwidth(
     }
 }
 
-void tr_bandwidth::phaseOne(std::vector<tr_peerIo*>& peer_array, tr_direction dir)
+void tr_bandwidth::phaseOne(std::vector<tr_peerIo*>& peers, tr_direction dir)
 {
     // First phase of IO. Tries to distribute bandwidth fairly to keep faster
     // peers from starving the others.
-    tr_logAddTrace(fmt::format("{} peers to go round-robin for {}", peer_array.size(), dir == TR_UP ? "upload" : "download"));
+    tr_logAddTrace(fmt::format("{} peers to go round-robin for {}", peers.size(), dir == TR_UP ? "upload" : "download"));
+
+    if (std::none_of(std::begin(peers), std::end(peers), [dir](auto const* peer) { return peer->has_bandwidth_left(dir); }))
+    {
+        return;
+    }
```

Comment:
Actually, not sure if needed. It possibly reduces performances to check has_bandwidth_left twice (here and inside the loop after).

Reply:
Removed in 554ae9d48e6402801c3c708355a569c81096010f